stable/search-prod/templates/addon-role.yaml

```diff
@@ -0,0 +1,31 @@
+# Copyright Contributors to the Open Cluster Management project
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: open-cluster-management:addons:search-collector
```

Comment:
Where is this role being used? We have a role with a similar name [here](https://github.com/stolostron/search-chart/blob/main/stable/search-prod/templates/collector-role.yaml) so I want to understand how these are different.

On a similar note, this chart always deploys a search-collector in the hub, so the add-on should never deploy another collector in the hub. This might be different from other add-ons because we always need the search-collector deployed in the hub, regardless of weather self-management is enabled or not.

Reply:
the search-collector on the managed cluster mounts a secret named search-collector-hub-kubeconfig to access the hub cluster  https://github.com/stolostron/search-operator/pull/99/files#diff-12fc76754e4bf7b3b41d80df778cc14b7c4543e1f7e35d28600c277dc7c31b14R60
This ClusterRole will be bond to the group in the hub-kubeconfig to grant permissions .
so we need a fixed name clusterRole, it will bind the clusterRole to the group here https://github.com/stolostron/search-operator/pull/99/files#diff-68d1bf5f2e78fc8ee5ca3ebd8893455f82700035a609ab7cdd2b28a828527197R92

Reply:
this clusterrole was deployed in cluster-lifecycle-chart on the hub https://github.com/stolostron/cluster-lifecycle-chart/blob/main/stable/cluster-lifecycle/templates/klusterlet-addon-search-role.yaml 
this migration will move it back to the search-chart.

Reply:
Thanks!